src/renderers/shared/stack/reconciler/ReactCompositeComponent.js

```diff
@@ -1107,7 +1107,9 @@ var ReactCompositeComponent = {
     if (__DEV__) {
       var componentName = component && component.getName ?
         component.getName() : 'a component';
-      warning(publicComponentInstance != null,
+      warning(
+        publicComponentInstance != null ||
+        component._compositeType !== CompositeTypes.StatelessFunctional,
```

Comment:
Oh, I understand this now. Different hidden class? Seems like we do this in many places anyway but thanks for explaining.